bftengine/include/bftengine/ControlStateManager.hpp

```diff
@@ -14,19 +14,47 @@
 #pragma once
 #include <optional>
 #include "IStateTransfer.hpp"
+#include "ReservedPages.hpp"
+#include "Serializable.h"
 
 namespace bftEngine {
 
-class ControlStateManager {
+class ControlStateManager : public ResPagesClient<ControlStateManager, 1, 1> {
  public:
   void setStopAtNextCheckpoint();
-  std::optional<uint64_t> getStopCheckpointToStopAt();
+  std::optional<bool> getStopCheckpointToStopAt();
 
   ControlStateManager(IStateTransfer& state_transfer);
   ControlStateManager& operator=(const ControlStateManager&) = delete;
   ControlStateManager(const ControlStateManager&) = delete;
+  ~ControlStateManager() {}
 
  private:
   IStateTransfer& state_transfer_;
+
+  static constexpr uint32_t size_of_reserved_pages_ = 512;
+  std::string scratchPage_;
+
+  // reserved page indexer
+  struct reservedPageIndexer {
+    uint32_t update_reserved_page_ = 0;
+  };
+
+  reservedPageIndexer reserved_pages_indexer_;
+};
+
+namespace controlStateMessages {
+
+// control state messages
+class StopAtNextCheckpointMessage : public concord::serialize::SerializableFactory<StopAtNextCheckpointMessage> {
```

Comment:
TODO @toly-kournik : subclass from Serializable

Reply:
Once #776 goes in I think we should use concord message format for these messages. Serializable is fine for local persistence, but it doesn't take into account endianness and is tightly coupled  to C++ type names. That should not hold up merging this though. The change can be made in a future commit.